code/go/0chain.net/smartcontract/minersc/miner.go

```diff
@@ -193,28 +193,20 @@ func (msc *MinerSmartContract) deleteNode(
 ) (*MinerNode, error) {
 	var err error
 	deleteNode.Delete = true
-	var nodeType spenum.Provider
 	switch deleteNode.NodeType {
```

Comment:
it's better to have
```go
if deleteNode.NodeType != NodeTypeMiner || eleteNode.NodeType != NodeTypeSharder {
    return nil, fmt.Errorf("unrecognised node type: %v", deleteNode.NodeType.String())
}
```

Reply:
I think the switch case is more readable, also we can simply add new valid type with a case instead of adding the long `|| deleteNode.NodeType != something`.